src/control.rs

```diff
@@ -0,0 +1,494 @@
+    flt_divider: SamplingClkDiv,
+    eev_divider: SamplingClkDiv,
+}
+
+impl HrTimOngoingCalibration {
+    /// SAFETY: Calibration needs to be done before calling this
+    unsafe fn init(self) {
+        let common = unsafe { &*HRTIM_COMMON::ptr() };
+
+        let Self {
+            #[cfg(feature = "stm32g4")]
+            adc_trigger1_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger2_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger3_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger4_postscaler,
+
+            #[cfg(feature = "stm32g4")]
+            adc_trigger5_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger6_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger7_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger8_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger9_postscaler,
+            #[cfg(feature = "stm32g4")]
+            adc_trigger10_postscaler,
+
+            flt_divider,
+            eev_divider,
+        } = self;
+
+        unsafe {
+            // Enable periodic calibration
+            // with f_hrtim at 170MHz, these settings leads to
+            // a period of about 6.2ms
+            common
+                .dllcr()
+                .modify(|_r, w| w.calrte().bits(0b00).cal().set_bit().calen().clear_bit());
+            common
+                .fltinr2()
+                .write(|w| w.fltsd().bits(flt_divider as u8));
+
+            common.eecr3().write(|w| w.eevsd().bits(eev_divider as u8));
+
+            #[cfg(feature = "stm32g4")]
+            common.adcps1().write(|w| {
+                w.adc1psc()
+                    .bits(adc_trigger1_postscaler as u8)
+                    .adc2psc()
+                    .bits(adc_trigger2_postscaler as u8)
+                    .adc3psc()
+                    .bits(adc_trigger3_postscaler as u8)
+                    .adc4psc()
+                    .bits(adc_trigger4_postscaler as u8)
+                    .adc5psc()
+                    .bits(adc_trigger5_postscaler as u8)
+            });
+
+            #[cfg(feature = "stm32g4")]
+            common.adcps2().write(|w| {
+                w.adc6psc()
+                    .bits(adc_trigger6_postscaler as u8)
+                    .adc7psc()
+                    .bits(adc_trigger7_postscaler as u8)
+                    .adc8psc()
+                    .bits(adc_trigger8_postscaler as u8)
+                    .adc9psc()
+                    .bits(adc_trigger9_postscaler as u8)
+                    .adc10psc()
+                    .bits(adc_trigger10_postscaler as u8)
+            });
+
+            // TODO: Adc trigger 5-10
+        }
+    }
+
+    pub fn wait_for_calibration(self) -> (HrTimCalibrated, FaultInputs, EevInputs) {
+        let common = unsafe { &*HRTIM_COMMON::ptr() };
+        while common.isr().read().dllrdy().bit_is_clear() {
```

Comment:
TODO: Should this even be done on the H7 since they do not have the DLL feature? Should this register field even exist in the pac?